api/utils.go

```diff
@@ -194,9 +194,29 @@ func tlsDialWithDialer(dialer *net.Dialer, network, addr string, config *tls.Con
 	// from the hostname we're connecting to.
 	if config.ServerName == "" {
 		// Make a copy to avoid polluting argument or default.
-		c := *config
+		c := &tls.Config{
+			Rand:                     config.Rand,
+			Time:                     config.Time,
+			Certificates:             config.Certificates,
+			NameToCertificate:        config.NameToCertificate,
+			GetCertificate:           config.GetCertificate,
+			RootCAs:                  config.RootCAs,
+			NextProtos:               config.NextProtos,
+			ServerName:               config.ServerName,
+			ClientAuth:               config.ClientAuth,
+			ClientCAs:                config.ClientCAs,
+			InsecureSkipVerify:       config.InsecureSkipVerify,
+			CipherSuites:             config.CipherSuites,
+			PreferServerCipherSuites: config.PreferServerCipherSuites,
+			SessionTicketsDisabled:   config.SessionTicketsDisabled,
+			SessionTicketKey:         config.SessionTicketKey,
+			ClientSessionCache:       config.ClientSessionCache,
+			MinVersion:               config.MinVersion,
+			MaxVersion:               config.MaxVersion,
+			CurvePreferences:         config.CurvePreferences,
+		}
```

Comment:
Actually this is integrated in `docker/docker/pkg/tlsconfig`. I'll switch it to that package after bumping to go1.7.1.